### ISSUE TITLE
Add rapidjson as submodule

### ISSUE THREAD

**Issue by cyenyxe** [vsuaste/testing_rapidJson#2]
Using the rapidjson repository as a Git submodule would simplify the build process. It can be done with the following command:

```
git submodule add git@github.com:miloyip/rapidjson.git
```

The header inclusion path should be added to the build commands in the README to reflect this change.

**Comment by vsuaste** [comment]
For me it worked `git submodule add https://github.com/miloyip/rapidjson.git`
I just added that part in the README.

**Comment by cyenyxe** [comment]
Yes, HTTPS is you are not going to submit anything to their repository.

By they way, since the submodule is already included in your repo, the compilation instructions do not need to cite the `git submodule add` command; the way to "clone" a submodule is using something like (I don't remember exactly) `git submodule update --init`. You can check by cloning the repository in a different folder and trying to recompile it.

**Comment by vsuaste** [comment]
I didn't want to have the submodule included in my repo. I was only checking if it works.
 I think it looks cleaner if I only mention the _adding submodule_ instruction in the README. 
what do you think?

**Comment by cyenyxe** [comment]
The submodule will not be stored in your repo, only a link will. Having them already "linked" lets people use your repo just by cloning it and populating the submodules with a single `git submodule update --init` instead of (potentially) multiple `git submodule add`, one per submodule.

And if they really want to know which are the dependencies, they easily can, because they are nicely displayed in the GitHub repo (see http://stackoverflow.com/a/16300781).

I only see advantages on having them explicitly added instead of just listed in the README :)
